modules/layers/src/text-layer/text-layer.js

```diff
@@ -180,17 +179,20 @@ export default class TextLayer extends CompositeLayer {
   // Text strings are variable width objects
   // Returns the index at the start of each string (every character is rendered by one instance)
   _updateText() {
-    const {data} = this.props;
+    const {data, characterSet} = this.props;
     const textBuffer = data.attributes && data.attributes.getText;
     let {getText} = this.props;
     let {startIndices} = data;
     let numInstances;
 
+    const autoCharacterSet = characterSet === 'auto' && new Set(DEFAULT_CHAR_SET);
+
     if (textBuffer && startIndices) {
       const {texts, characterCount} = getTextFromBuffer({
         ...(ArrayBuffer.isView(textBuffer) ? {value: textBuffer} : textBuffer),
         length: data.length,
-        startIndices
+        startIndices,
+        characterSet: autoCharacterSet
```

Comment:
I find it confusing how `autoCharacterSet` is passed in and then mutated by the `getTextFromBuffer` method. I think it would be better to explicitly return a modified version so it's obvious at this point that the method adds characters to the set

Reply:
This is an internal utility. Unnecessarily copying Set objects will add a performance hit.